### PR TITLE
[filtering] Add new move_forwards_from and move_backwards_from functions

### LINKED ISSUE #111: Some suggestions to enhance locating logics -'offset()' and 'resize()'
Hi @jstockwin ，

I am coming with some suggestions from my recent practice, would you please take a look? Thanks in advance.
1.  Adding 'offset()' and 'resize()' shortcut method of ElementList also Element:
For `ElementList.between()` method, with `inclusive` param, users are only able to select elements like `(e_1, ...e_n)` or `[e_1, ..., e_n]`. In first case, I want to select elements like `(e_1, ...e_n]` or `[e_1, ..., e_n)`(i.e.). In further case, I want to select elements like `(e_1-3, ...e_n+4)`, which requires elements outside the 'between' elements, maybe also requires `before`, `after` slices.
So I suggest giving users the freedom to easily choose elements:
'offset()': Giving e_0, I am able to choose e_-3 or e_3, so that I can easily locating elements between 2 offseted elements.
'resize()': Giving an element e_0, I am able to choose [e_0,...e_4].

The following is some demo of my actual usage without offset and resize, which troubles me.
```
    def locate_contents_between_locs(self, loc_pair_text,
            loc_pair_equal_or_contain=['equal','equal'],
            loc_pair_idx=[0,0],
            loc_pair_offset=[0,0],
            elements_scope=None,
            output_type=list,
            if_print_contents=False,
            ):
        if elements_scope == None:
            elements = self.doc.elements
        else:
            elements = elements_scope

        loc_pair = [None, None]
        for _ in range(2):
            if loc_pair_equal_or_contain[_] =='equal':
                loc_pair[_] = elements.filter_by_text_equal(loc_pair_text[_])[loc_pair_idx[_]]
            elif loc_pair_equal_or_contain[_] =='contain':
                loc_pair[_] = elements.filter_by_text_contains(loc_pair_text[_])[loc_pair_idx[_]]

            if loc_pair_offset[_] > 0:#inclusive [0, 1, 2...
                loc_pair[_] = elements.after(loc_pair[_], inclusive=True)[loc_pair_offset[_]]
            elif loc_pair_offset[_] < 0:#exclusive ...-2, -1]
                loc_pair[_] = elements.before(loc_pair[_], inclusive=False)[loc_pair_offset[_]]

        contents = elements.between(loc_pair[0], loc_pair[1])
        if if_print_contents:
            print(f'{contents}:{[c.text() for c in contents]}')
        
        return contents
```

### PR DESCRIPTION
**Pull request by jstockwin**
**Description**

Discussion in #111. Essentially, it was difficult to move through an element list starting at a given element. This makes that process both easier, and (I think) more efficient.

**Linked issues**

Closes #111

**Testing**

CI tests have been added.

**Checklist**

- [X] I have provided a good description of the change above
- [X] I have added any necessary tests
- [X] I have added all necessary type hints
- [X] I have checked my linting (`docker-compose run --rm lint`)
- [X] I have added/updated all necessary documentation (The docs generated form the docstrings are good enough here)
- [X] I have updated `CHANGELOG.md`, following the format from
      [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).

### CODE REVIEW

File: py_pdf_parser/filtering.py [hunk 1]
```diff
@@ -851,6 +852,71 @@ def remove_elements(self, *elements: "PDFElement") -> "ElementList":
...
+            count (int, optional): How many elements to move from `element`. The default
+                of 1 will move forwards by one element. Passing 0 will simply return the
+                element itself. You can also pass negative integers to move backwards.
+            capped (bool, optional): By default (False), if the count is high enough
+                that we try to move out of range of the list, an exception will be
+                raised. Passing `capped=True` will change this behaviour to instead
+                return the element at the start or end of the list.
+
+        Raises:
+            ElementOutOfRangeError: If the count large (or large-negative) enough that
```
Comment (paulopaixaoamaral) [inline py_pdf_parser/filtering.py:873]:
```suggestion
            ElementOutOfRangeError: If the count is large (or large-negative) enough that
```

File: py_pdf_parser/filtering.py [hunk 2]
```diff
@@ -851,6 +852,71 @@ def remove_elements(self, *elements: "PDFElement") -> "ElementList":
...
+        if new_index < 0 or new_index >= len(indexes):
+            # Out of range. We could simply catch the index error for large new_index,
+            # but we have to handle the negative case like this anyway, so might as well
+            # do both cases while we're at it.
+            if capped:
+                new_index = max(min(new_index, len(indexes) - 1), 0)
+                element_index = indexes[new_index]
+                return self.document._element_list[element_index]
+            raise ElementOutOfRangeError(
+                f"Requested element is {'before' if capped < 0 else 'after'} the start "
```
Comment (paulopaixaoamaral) [inline py_pdf_parser/filtering.py:887]: Shouldn't `new_index` be used here?
Also shouldn't it be `after the end of the ElementList`?

File: py_pdf_parser/filtering.py [hunk 3]
```diff
@@ -851,6 +852,71 @@ def remove_elements(self, *elements: "PDFElement") -> "ElementList":
...
+                of 1 will move backwards by one element. Passing 0 will simply return
+                the element itself. You can also pass negative integers to move
+                forwards.
+            capped (bool, optional): By default (False), if the count is high enough
+                that we try to move out of range of the list, an exception will be
+                raised. Passing `capped=True` will change this behaviour to instead
+                return the element at the start or end of the list.
+
+        Raises:
+            ElementOutOfRangeError: If the count large (or large-negative) enough that
```
Comment (paulopaixaoamaral) [inline py_pdf_parser/filtering.py:915]:
```suggestion
            ElementOutOfRangeError: If the count is large (or large-negative) enough that
```

File: py_pdf_parser/filtering.py (hunk 2 above)
Comment (jstockwin) [reply py_pdf_parser/filtering.py]: @paulopaixaoamaral good spot... see latest commit

**Pull request merged**